ui-lib/src/main/java/co/electriccoin/zcash/ui/common/serialization/addressbook/AddressBookKey.kt

```diff
@@ -0,0 +1,59 @@
+package co.electriccoin.zcash.ui.common.serialization.addressbook
+
+import cash.z.ecc.android.sdk.model.Account
+import cash.z.ecc.android.sdk.model.SeedPhrase
+import cash.z.ecc.android.sdk.model.ZcashNetwork
+import cash.z.ecc.android.sdk.tool.DerivationTool
+import com.google.crypto.tink.InsecureSecretKeyAccess
+import com.google.crypto.tink.aead.ChaCha20Poly1305Key
+import com.google.crypto.tink.subtle.Hkdf
+import com.google.crypto.tink.util.SecretBytes
+
+/**
+ * The long-term key that can decrypt an account's encrypted address book.
+ */
+class AddressBookKey(val key: SecretBytes) {
+    /**
+     * Derives a one-time address book encryption key.
+     *
+     * At encryption time, the one-time property MUST be ensured by generating a
+     * random 32-byte salt.
+     */
+    fun deriveEncryptionKey(salt: ByteArray): ChaCha20Poly1305Key {
+        assert(salt.size == ADDRESS_BOOK_SALT_SIZE)
+        val access = InsecureSecretKeyAccess.get()
+        val subKey = Hkdf.computeHkdf(
+            "HMACSHA256",
+            key.toByteArray(access),
+            salt,
+            null,
+            ADDRESS_BOOK_SALT_SIZE
```

Comment:
@Milan-Cerovsky this change is incorrect. This field of HKDF is not the length of `salt`, but the output length, i.e. the length of `subKey`.